### ISSUE TITLE
dom-index.js comment says in-order indexing.. but is it actually pre-order?

### ISSUE THREAD

**Issue by stepcut** [Matt-Esch/virtual-dom#298]
Hello,

I am trying to understand the code in:

https://github.com/Matt-Esch/vdom/blob/master/dom-index.js

The comment at the top says 'in-order tree indexing'. But I am not sure what that would even mean for an rose tree? It looks to me like it is more likely pre-order?

I could be very wrong!

**Comment by Matt-Esch** [comment]
Yes it is actually pre order (I believe this changed and the comment is stale). Effectively we are able to compute the next sibling index by adding the descendant count + 1 to the current sibling index. So root is 0, the first child is 1, the second child is `first child index + first child descendants + 1`
